### PR TITLE
Integration tests for Pytorch artifacts

### PR DESCRIPTION
**Pull request by jjmachan**
I have added the integration tests for PyTorch artifacts based on the commits form #726.

## Description

I've added a unit test for the Pytorch artefact were I create a simple model where the model is initialized with all *ones*. Hence the forward run of the model will give always give deterministic results. I also created a `run-keras-tests.py` script that installs tensorflow and performs the tests.

## Motivation and Context

This PR is based on #723 

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation
- [x] Test, CI, or build
- [ ] None

## Components (if applicable)
- [ ] BentoService (model packaging, dependency management, handler definition)
- [x] Model Artifact (model serialization, multi-framework support)
- [ ] Model Server (mico-batching, logging, metrics, tracing, benchmark, OpenAPI)
- [ ] YataiService (model management, deployment automation)
- [ ] Documentation

## Checklist:

- [x] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] My change requires a change in [bentoml/gallery](https://github.com/bentoml/gallery) example notebooks
- [ ] I have sent a pull request to [bentoml/gallery](https://github.com/bentoml/gallery) to make that change

### CODE REVIEW

**Comment by jjmachan** [comment]
There are some dependency issues with using PyTorch and I'm not sure what the best way to tackle this is

I created a new Travis job for PyTorch architecture and it requires PyTorch-1.5 so I added it to the script `run-keras-tests.py`. And now it seems to build fine. Also what exactly did you mean by optional? 

There is also a linting error due to torch and I'm not sure how to fix this one.

File: .travis.yml [hunk 1]
```diff
@@ -37,6 +37,11 @@ matrix:
       script:
         - ./travis/onnx_integration_tests.sh
 
+    - name: "Pytorch Artifact"
+      python: "3.6"
+      script:
+        - ./travis/run-keras-tests.sh
```
Comment (parano) [inline .travis.yml:43]: do you mean to name it `run-pytorch-tests.sh`?

File: tests/integration_tests/test_pytorch_model_artifact.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        torch.nn.init.ones_(self.linear.weight)
+
+    def forward(self, x):
+        x = self.linear(x)
+
+        return x
+
+
+@pytest.fixture()
+def pytorch_classifier_class():
```
Comment (parano) [inline tests/integration_tests/test_pytorch_model_artifact.py:26]: 👍

File: .travis.yml (hunk 1 above)
Comment (jjmachan) [reply .travis.yml:43]: yeah sry about that
in the issues page the script was named as such

File: tests/integration_tests/test_pytorch_model_artifact.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import pytest
+import pandas
+
+import torch
```
Comment (parano) [inline tests/integration_tests/test_pytorch_model_artifact.py:4]: just curious, why there is no need to add `pylint: disable=import-error` here?

File: .travis.yml [hunk 4]
```diff
@@ -37,6 +37,11 @@ matrix:
       script:
         - ./travis/onnx_integration_tests.sh
 
+    - name: "Pytorch Artifact"
+      python: "3.6"
+      script:
+        - ./travis/run-pytorch-tests.sh
```
Comment (parano) [inline .travis.yml:43]: could we follow the current naming convention here and name it `pytorch_integration_tests.sh` instead?  sorry that the issue description was a bit out dated

**Comment by parano** [comment]
Something worth exploring is to also test the API server in docker, for example: 1. use the saved_path to build a docker image, 2. start the API model server in a docker container, 3. send a test request to the API server

That will give us a lot more coverage I think. Although it might be a bit challenging to set up that in Travis, let me know if you’d be interested in trying out that as a follow-up

**Comment by jjmachan** [comment]
That seems interesting. I'll take a look at what I can do.

**Pull request merged**